chef/cookbooks/crowbar/templates/default/crowbar-delete-pacemaker-resources.sh.erb

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+#
+# Old pacemaker resources must be deleted in the old setup before they get created from upgraded node
+
+LOGFILE=/var/log/crowbar/node-upgrade.log
+mkdir -p "`dirname "$LOGFILE"`"
+exec >>"$LOGFILE" 2>&1
+
+echo "Executing $BASH_SOURCE"
+
+set -x
+
+<% if @use_ha %>
+
+echo "Stopping pacemaker resources..."
+
+
+for type in clone group ms primitive; do
+    for resource in $(crm configure show | awk "\$1 == \"$type\" && ! (\$2 ~ /drbd|stonith|vip-/) {print \$2}"); do
+        crm --wait resource stop $resource
+    done
+done
+
+echo "Deleting pacemaker resources..."
+
+for type in location clone group ms primitive; do
```

Comment:
> I haven't had any problems with order and colocation. Is it because it got deleted automatically with the resources they were referring too?

Possibly.  But the point is, we shouldn't do something which opens up the possibility of orphaned constraints, because we've already seen how that can give weird unwanted behaviour in the `g-haproxy` case.